torch/distributed/fsdp/_optim_utils.py

```diff
@@ -840,6 +774,56 @@ def _broadcast_unsharded_pos_dim_tensor_state(
     param_state[state_name] = unsharded_tensor
 
 
+def _rekey_sharded_optim_state_dict(
```

Comment:
This is different than the public `rekey_optim_state_dict()` since it is rekeying from an internal (private) `_OptimStateKey` to parameter ID.